spacy_llm/tasks/spancat.py

```diff
@@ -111,3 +112,13 @@ def assign_spans(
     ) -> None:
         """Assign spans to the document."""
         doc.spans[self._spans_key] = sorted(spans)  # type: ignore [type-var]
+
+    def scorer(
+        self,
+        examples: Iterable[Example],
+    ) -> Dict[str, Any]:
+        return spancat_score(
```

Comment:
Why `spancat_score` instead of `Scorer.score_spans`? (It seems odd that this isn't the same as the other tasks.)

Reply:
Because `spancat_score` implements the right getter out of the box. It's also the default scorer for spaCy's `spancat` component.

Reply:
But there are also default scorers for `ner` and `textcat` and `textcat_multilabel`?

Reply:
Maybe we should go ahead and do this as registered functions, which would reduce the potential future confusion here, I think.

Reply:
The problem is passing the settings in, which you don't really have access to in the config. Ugh.

Reply:
Ah, right. I used `get_ner_prf` for the NER task as you suggested and the scorers for `textcat`/`textcat_multilabel` only differ on the `multi_label` argument, so that seemed cleaner to use...

Reply:
Which settings do you not have access to in the config?

Reply:
> Ah, right. I used get_ner_prf for the NER task as you suggested and the scorers for textcat/textcat_multilabel only differ on the multi_label argument, so that seemed cleaner to use...

Sorry I was responding to a previous comment, my PR page was outdated...

Reply:
> Maybe we should go ahead and do this as registered functions, which would reduce the potential future confusion here, I think.

It would be nice to be able to customize the scoring function

Reply:
Added an internal ticket to follow up on this.

Reply:
The problem is that the scoring methods were originally written to be used from within the pipeline component itself, which could provide labels, settings, etc. If you move them to registered functions, then all those settings need to be provided to the registered function somehow. There's a terrible hack in spacy that passes all the settings in `pipe.cfg` from the component to the scorers, but that's not even possible from within the task here.